### PR TITLE
Add Enterprise SONiC "l3_acls" resource module

### PR DESCRIPTION
**Pull request by ArunSaravananBalachandran**
##### SUMMARY

- Add Enterprise SONiC 'l3_acls' resource module:
  - Provide data definitions in 'argspec'.
  - Implement 'facts' to gather facts for the module.
  - Implement 'config' to apply configurations for `merged`, `replaced`, `overridden` and `deleted` states.
- Define regression test cases to verify correct functionality of `l3_acls` resource module.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

- sonic_l3_acls

##### Test Results
- Regression test HTML report:
[l3_acls_regression_report.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/10871457/l3_acls_regression_report.pdf)

- Regression test detailed log:
[l3_acls_regression_terminal_output.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/10871458/l3_acls_regression_terminal_output.txt)

- Show running config and respective facts gathered:
[show_running_l3_acls_get_facts_output.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/10871460/show_running_l3_acls_get_facts_output.txt)

### CODE REVIEW

File: plugins/module_utils/network/sonic/config/l3_acls/l3_acls.py [hunk 1]
```diff
@@ -0,0 +1,716 @@
...
+            elif error_app_tag == 'update-not-allowed':
+                log = 'Creating ACLs with same name and different type not allowed'
+
+        if log:
+            response.update({u'log': log})
+            self._module.fail_json(msg=to_text(response), code=connection_error.code)
+        else:
+            self._module.fail_json(msg=str(connection_error), code=connection_error.code)
+
+    def _state_merged_overridden_replaced(self, want, have, state):
```
Comment (mingjunzhang2019) [inline plugins/module_utils/network/sonic/config/l3_acls/l3_acls.py:228]: Same comment as l2_acts, for same acl when merged, if "have" has more configuration attributes than "want",  are they deleted? or "merged" will not work for such case with this error message: "Cannot update existing sequence {0} of {1} ACL {2} with state merged. Please use state replaced or overridden."?

Comment (ArunSaravananBalachandran) [reply plugins/module_utils/network/sonic/config/l3_acls/l3_acls.py:228]: Yes, in state merged for an existing sequence number in a L3 ACL, if have and want attributes are not equal then the module execution will be aborted with the above error message.

Comment (mingjunzhang2019) [reply plugins/module_utils/network/sonic/config/l3_acls/l3_acls.py:228]: Thanks for explaination!

**Review by mingjunzhang2019 — APPROVED** [review]

**Comment by kerry-meyer** [comment]
This PR has conflicts that need to be resolved. Please resolve them and notify me. I will merge it after that.

**Pull request merged**